res/controllers/common-controller-scripts.js

```diff
@@ -14,15 +14,15 @@ String.prototype.toInt = function() {
 // ----------------- Function overloads ---------------------
 
 // Causes script print() calls to appear in the log file as well
-function print(string) {
+print = function(string) {
     engine.log(string);
 }
 
-function printObject(obj, maxdepth) {
+printObject = function(obj, maxdepth) {
     print(stringifyObject(obj, maxdepth));
 }
 
-function stringifyObject(obj, maxdepth, checked, prefix) {
+stringifyObject = function(obj, maxdepth, checked, prefix) {
```

Comment:
Please remove these new functions. They use new JS features that will not be supported before #1795 is merged and are off topic from this PR.

Reply:
If they happen to work it is just by coincidence that old JS interpreter tolerates them, but they may not really be working as you expect if you give them different input.

Reply:
Just curious, which JS features are you referring to? The function has served me well on many different inputs.

Reply:
My bad, I thought `try...catch` was a newer JS feature, but that has been around a long time. `const`, however, is not implemented in the current JS interpeter.

Reply:
Then I'll just replace const by var and done ^^